pkg/resources/secret_with_oauth_authorization_code_grant.go

```diff
@@ -0,0 +1,218 @@
+package resources
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/helpers"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/internal/logging"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/internal/provider"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/customdiff"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+var secretAuthorizationCodeGrantSchema = func() map[string]*schema.Schema {
+	secretAuthorizationCodeGrant := map[string]*schema.Schema{
+		"oauth_refresh_token": {
+			Type:        schema.TypeString,
+			Required:    true,
```

Comment:
I'm guessing this value should be sensitive

Reply:
Yes it should and i have added `Sensitive: true` to it